src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -499,39 +499,46 @@ public boolean hasSchemaEvolutionPermission(String tableName, String role) {
   }
 
   /**
-   * Alter table to add columns according to a map from columnNames to their types
+   * Execute queries to add columns or modify Iceberg table columns
    *
    * @param tableName the name of the table
-   * @param columnInfosMap the mapping from the columnNames to their infos
+   * @param query ADD COLUMN or ALTER SET DATA TYPE query
    */
   @Override
-  public void appendColumnsToTable(String tableName, Map<String, ColumnInfos> columnInfosMap) {
-    LOGGER.debug("Appending columns to snowflake table");
-    appendColumnsToTable(tableName, columnInfosMap, false);
+  public void evolveIcebergColumns(String tableName, String query) {
```

Comment:
see my comment in IcebergSchemaEvolutionService. The current implementation is just a pure proxy and executes arbitrary queries, it isn't in any way related to the iceberg. IMO we should generate the sql code here

Reply:
I wanted to pass a `String` query here so I don't expose List<IcebergoColumnTree> outside of the module.

"connectionService" is a better place to generate queries than a SchemaEvolutionService generates a schema etc...? Let the connection service just give me a connection and execute a statement that I want.

> it isn't in any way related to the iceberg
So maybe I'll just invoke 'executeStatement' method in Evolution service and generate log in evolution service.

Reply:
I generate query in connction service now.